File: peuler/ex4/ex4.c
#include<stdio.h>

int checkPal(int number) {
  int rev = 0, numberCopy = number;

  while (number != 0) {
	rev = rev * 10 + number % 10;
	number = (int) number / 10;
  }

  if (numberCopy == rev) return 1;

  return 0;
}

int main() {
  int factor1, factor2, maxPal = 0;

  for (factor1 = 999; factor1 >= 100; factor1--)
	for (factor2 = 999; factor2 >= 100; factor2--)
	  if (checkPal(factor1 * factor2))
		if (factor1 * factor2 > maxPal)
		  maxPal = factor1 * factor2;

  printf("%d is the largest 2-factor 3-digit palindrome.\n", maxPal);
  
  return 0;
}
  

File: peuler/ex4/ex4.py
def checkPal(number):
    rev = 0
    numberCopy = number

    while (number != 0):
        rev = rev * 10 + number % 10
        number = number // 10

    if (numberCopy == rev):
        return 1
    else:
        return 0

maxPal = 0
for factor1 in range(999, 100, -1):
    for factor2 in range(999, 100, -1):
        if (checkPal(factor1 * factor2)):
            if (factor1 * factor2 > maxPal):
                maxPal = factor1 * factor2

print(str(maxPal) + " is the largest 2-factor 3-digit palindrome.")
